### ISSUE TITLE
Database corruption on Fedora Linux 33

### ISSUE THREAD

**Issue by erthink** [erthink/libmdbx#144]
This is not a libmdbx, but Fedora:
- In a complex bash script FD=2 (stderr) got unexpectly closed  and then get used (i.e. returned from `open()`) by libmdbx inside `mdbx_env_open()`.
- then any application write to `stderr` leads to database corruption.

Bug was investigated by `strace` tool and then was caught by `assert(fd > 2)`.

**Comment by erthink** [comment]
The problem is in the time utility from Fedora 33.

```
$ bash -c "echo Ok"
Ok

$ bash -c "echo Ok >&2"
Ok

$ time bash -c "echo Ok"
Ok

real	0m0,006s
user	0m0,004s
sys	0m0,002s

$ time bash -c "echo Ok >&2"
Ok

real	0m0,006s
user	0m0,002s
sys	0m0,004s

$ /usr/bin/time bash -c "echo Ok"
Ok
0.00user 0.00system 0:00.00elapsed 60%CPU (0avgtext+0avgdata 3292maxresident)k
0inputs+0outputs (0major+170minor)pagefaults 0swaps
```

But(!)
```
$ /usr/bin/time bash -c "echo WTF >&2"
Command exited with non-zero status 1
0.00user 0.00system 0:00.00elapsed 100%CPU (0avgtext+0avgdata 3504maxresident)k
0inputs+0outputs (0major+182minor)pagefaults 0swaps

$ /usr/bin/time --version
time (GNU Time) UNKNOWN
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by David Keppel, David MacKenzie, and Assaf Gordon.
```

The same commands on Ubuntu:
```
$ /usr/bin/time bash -c 'echo ok >&2'
ok
0.00user 0.00system 0:00.00elapsed 100%CPU (0avgtext+0avgdata 3312maxresident)k
0inputs+0outputs (0major+141minor)pagefaults 0swaps

$ /usr/bin/time --version
GNU time 1.7
```

**Comment by erthink** [comment]
https://bugzilla.redhat.com/show_bug.cgi?id=1904650

**Comment by erthink** [comment]
Fixed in `time-1.9-12.fc33` Fedora package.
More over, libmdbx now have protection against such cases .
